client/src/routes/login.js

```diff
@@ -24,36 +25,50 @@ export default withStyles({
     teamToken: '',
     errors: {},
     disabledButton: false,
-    ctftimeToken: undefined
+    ctftimeToken: undefined,
+    pendingAuthToken: null,
+    pendingUserName: null,
+    pending: false
   }
 
-  componentDidMount () {
+  componentDidMount = async () => {
```

Comment:
This should not be async (the promise returned resolves immediately anyways, and componentDidMount should be void)